packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart

```diff
@@ -212,6 +285,12 @@ class DevServerRunner {
         if (shouldCacheSnapshot) _target.cacheLatestSnapshot();
         hasError = false;
       });
+
+      process.exitCode.then((code) async {
+        logger.detail('[process] exit($code)');
+        await _killServerProcess();
+        await stop(ExitCode.software);
```

Comment:
In such cases, we prioritize the local operation (kill-process, in this case) and then call stop, which is more generic to the whole class. This happens so we never miss cleaning the process because of conditions applied on `stop`.